docs/guides/architecture-scenarios.mdx

```diff
@@ -0,0 +1,93 @@
+
+Often-times such applications will also allow users to create personal accounts that are separate from other organizations. For example, GitHub allows users to create repositories under their own personal account or an organization they are part of.
+
+**TODO: diagram of B2B app with Clerk**
+
+The user pool for a multi-tenant, SaaS applications will generally fall into one of two categories:
+
+1. **Shared user-pool**: the application has a single pool of users. A user can create one account and belong to multiple organizations. The user can have separate roles in each organization.
+2. **Isolated user-pool**: each organization has its own pool of users. A user must create a separate account for each organization.
+
+<Callout type="info">
+Clerk supports the shared user-pool model for B2B scenarios which will be discussed in this section. The isolated user-pool model is more relevant in the Platforms scenario which will be discussed in the next section.
+</Callout>
+
+B2B SaaS applications with the following characteristics are well-supported with Clerk:
+
+- A single application deployment that serves multiple business customers (multi-tenant)
+- A shared user-pool model where a user can log in with a single account and belong to multiple organizations
+- Enabled connections are available to all users (i.e.: connections cannot be configured at the organization level)
+- The application may carry your own branding or some elements of your customer's branding
+- The application is served from a single domain (e.g.: `app.example.com`)
+
+### Integrating organizations with your application
+
+Clerk offers a number of building blocks to help integrate Organizations into your application:
+
+- The [`<OrganizationSwitcher />` component](https://clerk.com/docs/components/organization/organization-switcher) provides a way for your users to select which organization is active. The [`useOrganizationList()` hook](https://clerk.com/docs/organizations/custom-organization-switcher) can be used for more control.
+- The [`useOrganization()` hook](https://clerk.com/docs/references/react/use-organization) can be used to fetch the current, active organization.
+- The [`<Protect />` component](https://clerk.com/docs/components/protect) enables you to limit who can view certain pages based on their role. Additionally, Clerk exposes a number of helper functions [`auth()`](https://clerk.com/docs/references/nextjs/auth) and hooks [`useAuth()`](https://clerk.com/docs/references/react/use-auth#how-to-use-the-use-auth-hook) to check the user's authorization throughout your app and API endpoints.
+
+The organization's ID should be stored in your database, alongside each resource so that it can be used to filter and query the resources that should be rendered or returned according to the active organization.
```

Comment:
```suggestion
The organization's ID should be stored in your database alongside each resource so that it can be used to filter and query the resources that should be rendered or returned according to the active organization.
```